### DISCUSSION TITLE
`!`

### DISCUSSION THREAD

**Discussion by k4leg** [kovidgoyal/kitty#4446] [Q&A] [answered]
I use `alias !=sudo` in zsh (with omz and powerlevel10k), so in 0.24.0 it doesn't work (this constantly executes `sudo` and doesn't allow enter a command) because somewhere kitty use `!` with `builtin zle` (I found out this by substitution `alias !=echo`). Is there any way to fix this or is it still better for me to use another alias?
P. S.: my [zsh config](https://github.com/k4leg/dotfiles/blob/be35b2f3407fd8130a9db1896c28b0eea9b74ce8/dot_zshrc.tmpl) if it helps.

**Comment by page-down** [comment, marked as answer]
In my opinion, exclamation mark `!` are not appropriate to use as alias names. This is because exclamation mark is very commonly used in scripts. It is also easy to cause confusion.

**Comment by kovidgoyal** [comment]
Yes, ! is used as the Boolean operator ! in shell. You really should'nt be aliasing it.

**Comment by kovidgoyal** [comment]
Oh and @romkatv you wanted to be pinged if someone came up with new and creative ways to break the shell integration script. It's a bit weird though since I thought alias expansion was disabled in autoloaded code via -Uz

I suppose -U only suppresses aliases during loading not execution.

**Comment by romkatv** [comment]
That bug's on me. Sorry about that. https://github.com/kovidgoyal/kitty/pull/4449 should fix it.

Aliases are indeed disabled in kitty-integration but they are enabled when deferred init is patching functions.
